tests/python/helpers.py

```diff
@@ -1,45 +1,197 @@
 
-from concurrent.futures import ThreadPoolExecutor
+import kin
+import kin_base
+
+NETWORK_NAME = 'LOCAL'
+MIN_FEE = 100
+MAX_OPS = 100
+TX_SET_SIZE = 500
+
+
+def root_account_seed(passphrase: str) -> str:
+    """Return the root account seed based on the given network passphrase."""
+    network_hash = sha256(passphrase.encode()).digest()
+    return kin_base.Keypair.from_raw_seed(network_hash).seed().decode()
 
 
 def derive_root_account(passphrase):
-    """Return the keypair of the root account, based on the network passphrase"""
+    """Return the keypair of the root account, based on the network passphrase."""
     network_hash = sha256(passphrase.encode()).digest()
     seed = BaseKeypair.from_raw_seed(network_hash).seed().decode()
     return Keypair(seed)
 
 
-async def send_txs(txs):
-    with ThreadPoolExecutor(max_workers=len(txs)) as executor:
-        loop = asyncio.get_running_loop()
-        promises = [loop.run_in_executor(executor, tx.submit) for tx in txs]
+async def create_accounts(env: Environment, channels: List[Builder], root_account: BaseKeypair, starting_balance, horizon_addresses: List[str] = []):
+    """Asynchronously create accounts and return a transaction Builder instance for each account.
+
+    Each Builder instance uses a different Horizon endpoint if given a non-empty list.
+    """
+    accounts_to_create_num = len(channels) * MAX_OPS
+    logging.info('creating %d accounts', accounts_to_create_num)
+
+    accounts_to_create = [Keypair() for _ in range(accounts_to_create_num)]
+    xdrs = []
+    for i, channel in enumerate(channels):
+        for kp in accounts_to_create[i*MAX_OPS : i*(MAX_OPS) + MAX_OPS]:
+            # use the root account as the source of the operations
+            #
+            # NOTE channels need a starting balance since they need to pay fees because they are not prioritized
+            channel.append_create_account_op(source=root_account.address(),
+                                             destination=kp.public_address,
+                                             starting_balance=str(starting_balance))
+
+        # sign with channel and root account
+        channel.sign(secret=channel.keypair.seed().decode())
+        if root_account.seed().decode() != channel.keypair.seed().decode():
+            channel.sign(secret=root_account.seed().decode())
+        xdrs.append(channel.gen_xdr())
+
+        channel.next()
+
+    horizon_uri = channels[0].horizon.horizon_uri
+    await send_txs(horizon_uri, xdrs, expected_statuses=[200])
+
+    # fetch sequence numbers asynchronously for all created accounts
+    sequences = await get_sequences(horizon_uri, [a.public_address for a in accounts_to_create])
+
+    # create tx builders with up-to-date sequence number
+    builders = []
+    horizons = ([Horizon(horizon_uri=addr, user_agent=SDK_USER_AGENT) for addr in horizon_addresses]
+                if horizon_addresses
+                else [horizon_uri])
```

Comment:
if there are no horizon addresses you should create a list with a single horizon instance, but you are creating a list with a single string (the horizon endpoint)